feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts

```diff
@@ -46,106 +57,97 @@ export class ConfigureCartEntryComponent {
+      }
+      case AbstractOrderType.QUOTE: {
+        return CommonConfigurator.OwnerType.QUOTE_ENTRY;
+      }
+      case AbstractOrderType.SAVED_CART: {
+        return CommonConfigurator.OwnerType.SAVED_CART_ENTRY;
+      }
+      default: {
+        return CommonConfigurator.OwnerType.CART_ENTRY;
+      }
+    }
   }
 
   /**
+   * @deprecated Use retrieveEntityKey instead
    * Verifies whether the cart entry has an order code, retrieves a composed owner ID
    * and concatenates a corresponding entry number.
    *
    * @returns - an entry key
    */
   getEntityKey(): string {
-    return this.getEntityKeyInternal(true);
+    const entryNumber = this.cartEntry.entryNumber;
+    if (entryNumber === undefined) {
+      throw new Error('No entryNumber present in entry');
+    }
+
+    return this.cartEntry.orderCode
+      ? this.commonConfigUtilsService.getComposedOwnerId(
+          this.cartEntry.orderCode,
+          entryNumber
+        )
+      : entryNumber.toString();
   }
 
   /**
    * Verifies whether the cart entry has an order code, retrieves a composed owner ID
    * and concatenates a corresponding entry number.
-   * @param cartCode Code of active cart
+   *
    * @returns - an entry key
    */
-  getEntityKeyKnowingActiveCart(cartCode: string): string {
-    return this.getEntityKeyInternal(this.isActiveCart(cartCode));
-  }
-
-  protected isActiveCart(cartCode: string): boolean {
-    return cartCode === this.cartEntry.savedCartCode;
-  }
-
-  protected getOwnerTypeInternal(
-    readOnlyCartIsActive: boolean
-  ): CommonConfigurator.OwnerType {
-    if (this.cartEntry.orderCode) {
-      return CommonConfigurator.OwnerType.ORDER_ENTRY;
-    } else if (this.readOnly && this.cartEntry.quoteCode) {
-      return CommonConfigurator.OwnerType.QUOTE_ENTRY;
-    } else if (this.readOnly && !readOnlyCartIsActive) {
-      return CommonConfigurator.OwnerType.SAVED_CART_ENTRY;
-    } else {
-      return CommonConfigurator.OwnerType.CART_ENTRY;
-    }
-  }
-
-  protected getEntityKeyInternal(readOnlyCartIsActive: boolean): string {
+  retrieveEntityKey(abstractOrderData: AbstractOrderKey): string {
+    const orderType = abstractOrderData.type;
+    const ownerDocumentIdNeeded: boolean =
+      orderType === AbstractOrderType.ORDER ||
+      orderType === AbstractOrderType.QUOTE ||
+      orderType === AbstractOrderType.SAVED_CART;
     const entryNumber = this.cartEntry.entryNumber;
     if (entryNumber === undefined) {
       throw new Error('No entryNumber present in entry');
     }
-    let code;
-    if (this.cartEntry.orderCode) {
-      code = this.cartEntry.orderCode;
-    } else if (this.readOnly && this.cartEntry.quoteCode) {
-      code = this.cartEntry.quoteCode;
-    } else if (this.readOnly && !readOnlyCartIsActive) {
-      code = this.cartEntry.savedCartCode;
-    }
-    return code
-      ? this.commonConfigUtilsService.getComposedOwnerId(code, entryNumber)
+    return ownerDocumentIdNeeded
```

Comment:
If `ownerDocumentIdNeeded` is renamed to `ownerIdRequired`, then it will be clearer when getConfiguratorOwnerId() method is called in case `ownerIdRequired is **true**